### PR TITLE
[Improve][Connector-V2] Improve http connector

### PR DESCRIPTION
**Pull request by hailin0**
## Purpose of this pull request

* Support read json array
* Support `SimpleTextSchema`
* Support retry for request http server
* Support request http interval in stream mode

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by hailin0** [comment]
@TyrantLucifer PTAL

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonToRowConverters.java [hunk 1]
```diff
@@ -107,6 +114,31 @@ private JsonToRowConverter createNotNullConverter(SeaTunnelDataType<?> type) {
                 return this::convertToBytes;
             case DECIMAL:
                 return createDecimalConverter((BasicType<BigDecimal>) type);
+            case ARRAY:
+                ArrayType arrayType = (ArrayType) type;
```
Comment (Hisoka-X) [inline seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonToRowConverters.java]: Please extract method, same with other type

Comment (hailin0) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/JsonToRowConverters.java]: done

**Review by TyrantLucifer — APPROVED** [review]
LGTM

**Review by ashulin — APPROVED** [review]
LGTM

**Pull request merged**